atom/browser/api/atom_api_web_contents.cc

```diff
@@ -861,6 +861,12 @@ void WebContents::Observe(int type,
   }
 }
 
+void WebContents::BeforeUnloadDialogCancelled() {
+  if (deferred_load_url_.id) {
```

Comment:
`BeforeUnloadDialogCancelled` will also be called for any cancellation inside iframes, we do not want to clear the defer load params in that situation, it would have been better if there was an argument that contained the frame id. The only way now is to check if there is an pending entry or not. This observer will be called only after clearing navigation controller states for this `webContents`, so we can be sure any pending entry would have been cleared.

```
if (deferrer_load_url_.id && !controller.GetPendingEntry())
```